http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/GceRuntimeMonitorInterp.scala

```diff
@@ -0,0 +1,698 @@
+    // Otherwise, when a cluster is resumed and transitions from Starting to Running, we get stuck
+    // in that state - at least with the way HttpJupyterDAO.isProxyAvailable works
+    for {
+      ctx <- ev.ask
+      imageTypes <- if (toolsToCheck.nonEmpty) //If toolsToCheck is defined, we've checked some tools already, and only checking tools that haven't been available yet
+        F.pure(toolsToCheck)
+      else
+        dbRef
+          .inTransaction { //If toolsToCheck is defined, then we haven't check any tools yet. Hence retrieve all tools from database
+            for {
+              _ <- clusterQuery.updateClusterHostIp(runtimeAndRuntimeConfig.runtime.id, Some(ip), ctx.now)
+              images <- clusterImageQuery.getAllForCluster(runtimeAndRuntimeConfig.runtime.id)
+            } yield images.toList.map(_.imageType)
+          }
+      availableTools <- imageTypes.traverseFilter { imageType =>
+        RuntimeContainerServiceType.imageTypeToRuntimeContainerServiceType
+          .get(imageType)
+          .traverse(
+            _.isProxyAvailable(runtimeAndRuntimeConfig.runtime.googleProject,
+                               runtimeAndRuntimeConfig.runtime.runtimeName).map(b => (imageType, b))
+          )
+      }
+      res <- availableTools match {
+        case a if a.forall(_._2) =>
+          readyRuntime(runtimeAndRuntimeConfig, ip, monitorContext)
+        case a =>
+          val toolsStillNotAvailable = a.collect { case x if x._2 == false => x._1 }
+          checkAgain(
+            monitorContext,
+            runtimeAndRuntimeConfig,
+            Some(s"Services not available: ${toolsStillNotAvailable}"),
+            Some(ip),
+            toolsStillNotAvailable
+          )
+      }
+    } yield res
+
+  private def checkAgain(monitorContext: MonitorContext,
+                         runtimeAndRuntimeConfig: RuntimeAndRuntimeConfig,
+                         message: Option[String],
+                         ip: Option[IP] = None,
+                         toolsToCheck: List[RuntimeImageType] = List.empty // If empty, then check all tools; if defined, then only check specified tools
+  ): F[CheckResult] = {
+    for {
+      now <- nowInstant
+      implicit0(traceId: ApplicativeAsk[F, AppContext]) = ApplicativeAsk.const[F, AppContext](AppContext(monitorContext.traceId, now))
+      timeElapsed = (now.toEpochMilli - monitorContext.start.toEpochMilli).millis
+      res <- config.monitorStatusTimeouts.get(runtimeAndRuntimeConfig.runtime.status) match {
+        case Some(timeLimit) if timeElapsed > timeLimit =>
+          for {
+            _ <- logger.info(
+              s"Detected that ${runtimeAndRuntimeConfig.runtime.projectNameString} has been stuck in status ${runtimeAndRuntimeConfig.runtime.status} too long."
+            )
+            r <- // Take care not to Error out a cluster if it timed out in Starting status
```

Comment:
scalafmt?

Reply:
mmm...yea, this is weird formatting 🤷‍♀️